mitodl/open-discussions
# add comment permalink

#### What are the relevant tickets?

closes #272 

#### What's this PR do?

This adds a 'permalink' to each comment. Basically, we have to do the following:

- add a new API endpoint for GET-ing a single comment and it's children
- modify the `PostPage` to look for a comment parameter in the URL and, when present, show some different UI and fetch just that comment instead of all comments
- make a slight change to the comments reducer to support handling a top-level comment whose parent we do not have (which we need if the comment we're 'focused' on is a nested comment, since it's parents will not be returned from the API).

I think those are the major changes.

#### How should this be manually tested?

Every comment should have a little 'permalink' button. If you click on it you should see a special UI, basically a little 'hey you're looking at one comment' thingy and the post display should collapse a little bit. You should be able to click back to all the comments. Looking at all the comments for a post should work as normal.

Here's what it looks like:

![beepboop](https://user-images.githubusercontent.com/6207644/35059553-8a315b8e-fb89-11e7-945e-0ddc764855f3.png)
